test/custom-parser.3.test.js

```diff
@@ -74,14 +71,15 @@ test('contentTypeParser should add a custom parser with RegExp value', t => {
           'Content-Type': 'application/vnd.test+json'
         }
       }, (err, response, body) => {
-        t.error(err)
-        t.equal(response.statusCode, 200)
-        t.same(body.toString(), JSON.stringify({ hello: 'world' }))
+        t.assert.ifError(err)
+        t.assert.strictEqual(response.statusCode, 200)
+        t.assert.deepStrictEqual(body.toString(), JSON.stringify({ hello: 'world' }))
       })
     })
 
-    t.test('in OPTIONS', t => {
+    await t.test('in OPTIONS', (t) => {
```

Comment:
we would still need to add in `done` and call after last assert
```suggestion
    await t.test('in POST', (t, done) => {
```

Reply:
Done! see here: [8ccbf71](https://github.com/fastify/fastify/pull/5903/commits/8ccbf71c4bfddae19cd2e07926a58ade1f610813)